### DISCUSSION TITLE
Send The Images to a Friend

### DISCUSSION THREAD

**Discussion by Deepak-Beniya** [OE4T/tegra-demo-distro#290] [Q&A] [answered]
Hlo Everyone ,
Just a short question that one of my friend have Jetson Orin NX board.
I have compiled the Yocto source by following the README , Now what images I need to send to my friend  ?
How will he flash and Where are the Images locations ? I mean Path . 
I need all zImage, dtb , rootfs , primary bootloader, secondary bootloader .

Or is there any one single file like  .sdcard  or  .wic  or  tar.gz  file that includes  all the images ?

Thanks and Regards
Deepak Kumar Beniya

**Comment by dwalkes** [comment]
See the page I shared with you previously and the tegraflash package described there at https://github.com/OE4T/meta-tegra/wiki/Flashing-the-Jetson-Dev-Kit#building-a-tegraflash-package

**Reply by Deepak-Beniya** [reply]
I went through it but what about the questions I have asked ?
It will be helpfull if you answer one by one.

**Reply by Deepak-Beniya** [reply]
Image
Image--5.10.120+git0+76678311c1-r0-jetson-orin-nano-devkit-20240213060335.bin
Image.gz
Image.gz--5.10.120+git0+76678311c1-r0-jetson-orin-nano-devkit-20240213060335.bin
Image.gz-jetson-orin-nano-devkit.bin
Image-jetson-orin-nano-devkit.bin
modules--5.10.120+git0+76678311c1-r0-jetson-orin-nano-devkit-20240213060335.tgz
modules-jetson-orin-nano-devkit.tgz
tegra234-p3767-0003-p3768-0000-a0--5.10.120+git0+76678311c1-r0-jetson-orin-nano-devkit-20240213060335.dtb
tegra234-p3767-0003-p3768-0000-a0--5.10.120+git0+76678311c1-r0-jetson-orin-nano-devkit-20240213060335.dtb.signed
tegra234-p3767-0003-p3768-0000-a0.dtb
tegra234-p3767-0003-p3768-0000-a0.dtb.signed
tegra234-p3767-0003-p3768-0000-a0-jetson-orin-nano-devkit.dtb
tegra234-p3767-0003-p3768-0000-a0-jetson-orin-nano-devkit.dtb.signed
tegra234-p3767-overlay.dtbo
tegra234-p3768-overlay.dtbo



from these which file should i use ? 
I guess "modules--5.10.120+git0+76678311c1-r0-jetson-orin-nano-devkit-20240213060335.tgz" this may be the right file because its 520MB .

if i will send this zip file through Google Drive  to my friend then he will be abel to flash in board ??

**Reply by madisongh** [reply]
None of those are complete filesystem images; they look like the artifacts from just a kernel build.  The recipes to build for complete images are mentioned in the [Images section of the README](https://github.com/OE4T/tegra-demo-distro?tab=readme-ov-file#images).  The resulting artifact will be named as described in [this section of the wiki page Dan referred you to](https://github.com/OE4T/meta-tegra/wiki/Flashing-the-Jetson-Dev-Kit#building-a-tegraflash-package).

**Reply by Deepak-Beniya** [reply]
Once again i have compiled the demo-full-image for jetson orion nano board . 
I have sent the folder  to my friend .
 build/tmp/deploy/images/jetson-orin-nano-devkit/    this folder i have sent to him and he has the jetson orin nano board.
 
 Could you tell me what he should do to flash the images.
 I have went through the flashing document https://github.com/OE4T/meta-tegra/wiki/Flashing-the-Jetson-Dev-Kit#building-a-tegraflash-package .
 
 But it is dificult to inderstand the readme . Please write the steps in here that would be helpfull. @madisongh @dwalkes 
 
 I hope you could understand my problem.
 
 These are the files which the folder contains.



demo-image-full-jetson-orin-nano-devkit-20240213043354.rootfs.ext4
demo-image-full-jetson-orin-nano-devkit-20240213043354.rootfs.manifest
demo-image-full-jetson-orin-nano-devkit-20240213043354.tegraflash.tar.gz
demo-image-full-jetson-orin-nano-devkit-20240213043354.testdata.json
demo-image-full-jetson-orin-nano-devkit.ext4
demo-image-full-jetson-orin-nano-devkit.manifest
demo-image-full-jetson-orin-nano-devkit.tegraflash.tar.gz
demo-image-full-jetson-orin-nano-devkit.testdata.json
Image
Image--5.10.120+git0+76678311c1-r0-jetson-orin-nano-devkit-20240213043354.bin
Image.gz
Image.gz--5.10.120+git0+76678311c1-r0-jetson-orin-nano-devkit-20240213043354.bin
Image.gz-jetson-orin-nano-devkit.bin
Image-jetson-orin-nano-devkit.bin
L4TConfiguration.dtbo
L4TConfiguration-rcmboot.dtbo
modules--5.10.120+git0+76678311c1-r0-jetson-orin-nano-devkit-20240213043354.tgz
modules-jetson-orin-nano-devkit.tgz
tegra234-p3767-0003-p3768-0000-a0--5.10.120+git0+76678311c1-r0-jetson-orin-nano-devkit-20240213043354.dtb
tegra234-p3767-0003-p3768-0000-a0--5.10.120+git0+76678311c1-r0-jetson-orin-nano-devkit-20240213043354.dtb.signed
tegra234-p3767-0003-p3768-0000-a0.dtb
tegra234-p3767-0003-p3768-0000-a0.dtb.signed
tegra234-p3767-0003-p3768-0000-a0-jetson-orin-nano-devkit.dtb
tegra234-p3767-0003-p3768-0000-a0-jetson-orin-nano-devkit.dtb.signed
tegra234-p3767-overlay.dtbo
tegra234-p3768-overlay.dtbo
tegra-espimage-jetson-orin-nano-devkit-20240213043354.rootfs.esp
tegra-espimage-jetson-orin-nano-devkit-20240213043354.rootfs.manifest
tegra-espimage-jetson-orin-nano-devkit-20240213043354.testdata.json
tegra-espimage-jetson-orin-nano-devkit.esp
tegra-espimage-jetson-orin-nano-devkit.manifest
tegra-espimage-jetson-orin-nano-devkit.testdata.json
tegra-initrd-flash-initramfs-jetson-orin-nano-devkit-20240213043354.rootfs.cpio.gz.cboot
tegra-initrd-flash-initramfs-jetson-orin-nano-devkit-20240213043354.rootfs.manifest
tegra-initrd-flash-initramfs-jetson-orin-nano-devkit-20240213043354.testdata.json
tegra-initrd-flash-initramfs-jetson-orin-nano-devkit.cboot
tegra-initrd-flash-initramfs-jetson-orin-nano-devkit.cpio.gz.cboot
tegra-initrd-flash-initramfs-jetson-orin-nano-devkit.manifest
tegra-initrd-flash-initramfs-jetson-orin-nano-devkit.testdata.json
tegra-minimal-initramfs-jetson-orin-nano-devkit-20240213043354.bl_only.bup-payload
tegra-minimal-initramfs-jetson-orin-nano-devkit-20240213043354.kernel_only.bup-payload
tegra-minimal-initramfs-jetson-orin-nano-devkit-20240213043354.rootfs.cpio.gz
tegra-minimal-initramfs-jetson-orin-nano-devkit-20240213043354.rootfs.cpio.gz.cboot
tegra-minimal-initramfs-jetson-orin-nano-devkit-20240213043354.rootfs.manifest
tegra-minimal-initramfs-jetson-orin-nano-devkit-20240213043354.testdata.json
tegra-minimal-initramfs-jetson-orin-nano-devkit.bl_only.bup-payload
tegra-minimal-initramfs-jetson-orin-nano-devkit.cboot
tegra-minimal-initramfs-jetson-orin-nano-devkit.cpio.gz
tegra-minimal-initramfs-jetson-orin-nano-devkit.cpio.gz.cboot
tegra-minimal-initramfs-jetson-orin-nano-devkit.kernel_only.bup-payload
tegra-minimal-initramfs-jetson-orin-nano-devkit.manifest
tegra-minimal-initramfs-jetson-orin-nano-devkit.testdata.json
tos-jetson-orin-nano-devkit-35.4.1-r0.img
tos-jetson-orin-nano-devkit.img
uefi_jetson.bin

**Reply by ichergui** [reply]
Hi @Deepak-Beniya 

Could you please elaborate about the difficulties you are facing with the wiki page ?

**Reply by Deepak-Beniya** [reply]
@ichergui 
I don't have the board with me. I have compiled everything and given to my friend. 
doflash.sh script is not present. 
Could you tell me the steps how can I flash it.

**Reply by madisongh** [reply, marked as answer]
As mentioned on the wiki page [here](https://github.com/OE4T/meta-tegra/wiki/Flashing-the-Jetson-Dev-Kit#building-a-tegraflash-package), it's the `tegraflash.tar.gz` file for the image that is needed.  Unpack the tarball as described on the wiki page [here](https://github.com/OE4T/meta-tegra/wiki/Flashing-the-Jetson-Dev-Kit#unpacking-the-tegraflash-package), and you should find all of the files needed for flashing the device.
